### PR TITLE
Add gelu gradient for pytorch

### PR DESCRIPTION
**Pull request by xiaomengy**
Summary: Add gelu gradient for pytorch

Differential Revision: D15589816

### CODE REVIEW

File: test/test_nn.py [hunk 1]
```diff
@@ -6067,16 +6067,19 @@ def _gelu_ref(X):
...
-                X = torch.rand(n, m, dtype=dtype)
+                X = torch.rand(n, m, dtype=dtype, requires_grad=True)
             else:
-                X = torch.rand(n, m, dtype=dtype)[:, ::2]
+                X = torch.rand(n, m, dtype=dtype, requires_grad=True)[:, ::2]
             res = F.gelu(X)
-            ref = _gelu_ref(X.numpy())
+            ref = _gelu_ref(X.detach().numpy())
             self.assertEqual(res, ref)
+            gradcheck(F.gelu, [X], eps=1e-4, atol=1e-3)
```
Comment (soumith) [inline test/test_nn.py]: this looks suspicious. gradchecks happen in double precision, so a tolerance of 1e-3 looks really high and a custom eps is usually not needed. Any idea what is going on? Can you check some sample inputs to inspect.

**Review by soumith — APPROVED** [review]
please check inline comment and provide resolution on what's up with the tolerance adjustment. Once you get clarity into that, and verify it's not a bug, do land.

Things reviewed in the diff:
- MKL and non-MKL implementations match in formula
- CUDA and CPU implementation match in formula

Things not reviewed in the diff:
- gradient formula is correct (relying on gradcheck to say it's right)

File: test/test_nn.py (hunk 1 above)
Comment (BIT-silence) [reply test/test_nn.py]: This part just came from testing the numerical stability for gradchecks. Atol has been removed and now it is using default value.
